egs/callhome_diarization/v1/diarization/convert_VB_model.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python
```

Comment:
In my opinion, it would be a better design to move the functionality of this script into VB_resegmentation.py and remove both this script and the bash script convert_VB_model.sh. That way, VB_resegmentation.py would take Kaldi models directly as input, and the conversion into numpy objects would happen internally after loading the objects.

In general, I think we should prefer that Kaldi scripts have interfaces that use Kaldi objects. The idea that you need to first convert a Kaldi object into a non-Kaldi object and dump it to disk before loading it into another Kaldi script doesn't feel right to me. Of course if there's a reason we really have to do it this way, it can be kept as is, but it doesn't seem necessary to me.

Reply:
Yes, I feel the same way. It looks ugly for me too. Do you know any convenient way to load Kaldi objects(diagonal UBM and i-vector extractor in my case) directly as python dictionary? Perhaps that is much elegant. So I don't need to convert it to a text file first, which is extremely ugly. I guess Dan @danpovey might have good ideas about that?

Reply:
I suspect the amount of work required to change this would be disproportionate to the benefit.
This VB resegmentaion may  not even be something we rely on in future.  So I'd say, just let sleeping dogs lie for now.

Reply:
Okay, thanks Dan.